service/history/reset/resetter.go

```diff
@@ -632,3 +619,56 @@ func (r *workflowResetterImpl) getPaginationFn(
 		return paginateItems, token, nil
 	}
 }
+
+func (r *workflowResetterImpl) closePendingDecisionTask(
+	resetMutableState execution.MutableState,
+	baseRunID string,
+	resetRunID string,
+	baseLastEventVersion int64,
+	resetReason string,
+) (execution.MutableState, error) {
+
+	if decision, ok := resetMutableState.GetInFlightDecision(); ok {
+		// reset workflow has decision task start
+		if len(resetMutableState.GetPendingChildExecutionInfos()) > 0 {
+			return nil, &shared.BadRequestError{
+				Message: fmt.Sprintf("Can not only reset workflow with pending child workflows"),
+			}
+		}
+		_, err := resetMutableState.AddDecisionTaskFailedEvent(
+			decision.ScheduleID,
+			decision.StartedID,
+			shared.DecisionTaskFailedCauseResetWorkflow,
+			nil,
+			execution.IdentityHistoryService,
+			resetReason,
+			"",
+			baseRunID,
+			resetRunID,
+			baseLastEventVersion,
+		)
+		if err != nil {
+			return nil, err
+		}
+	} else if decision, ok := resetMutableState.GetPendingDecision(); ok {
+		if ok {
+			//reset workflow has decision task schedule
+			if len(resetMutableState.GetPendingChildExecutionInfos()) > 0 {
```

Comment:
Nit: maybe move it to the beginning of the method.

Reply:
updated